src/smc_sagews/smc_sagews/sage_salvus.py

```diff
@@ -3612,6 +3612,40 @@ def help(*args, **kwds):
 - **License information:** For license information about Sage and its components, enter `license()`."""%sage.version.version
         salvus.md(s)
 
-
 # Import the jupyter kernel client.
 from sage_jupyter import jupyter
+
+# search_src
+import os
+def search_src(str):
+    r"""
+    Get file names resulting from git grep of smc repo
+
+    INPUT:
+
+    - ``str`` -- string expression to search for; will be quoted
+
+    OUTPUT:
+
+    Interact selector of matching filenames. Choosing one causes its
+    contents to be shown in salvus.code() output.
+    """
+    cmd = "cd /projects/sage/sage/src/sage;git grep -il '" + str + "'"
```

Comment:
and another idea for grep: prepend it with `timeout 5 grep -il ...` such that it times out after 5 secs and you still get the output. Maybe 10 secs? IDK.